sabnzbd/sabnzbd
# New SSL approach

## **Of course this is not for 1.1.0, but 1.2.0**

After discussion in #628, this is my attempt at removing our dependency on pyOpenSSL and using native ```ssl``` module.

- The user can no longer select what protocol to use, it will always use the best possible. The used protocol and cipher will now be logged after connection is completed.
- There is a new Switch, where you can select the level of verification: Off, Normal (only require a good certificate) and Strict (also force hostname check). 
- To test:
```news.surfino.com``` which only works when it's set to Off. 
```eunews.frugalusenet.com``` won't work in Strict-mode.
- Updated CherryPy to version 8.1.0 to use ```ssl``` instead of PyOpenSSL as SSL provider. They have patched almost everything we did, but I still had to make 1 custom patch, see: https://github.com/cherrypy/cherrypy/issues/1497
- 301 redirect HTTP->HTTPS on the HTTPS port doesn't work yet, but that's for later.

So, what do you think? 🐹
